packages/amplify-graphql-transformer-core/src/migration/migrators/key/index.ts

```diff
@@ -0,0 +1,69 @@
+import { createArgumentNode, createDirectiveNode } from "../generators";
+
+export function isPrimaryKey(directive: any) {
+    if (directive.name.value === 'key'
+        && directive.arguments.find((a: any) => a.name.value === "fields")
+        && !directive.arguments.find((a: any) => a.name.value === "name")) {
+        return true;
+    }
+    return false;
+}
+
+export function migratePrimaryKey(node: any, directive: any) {
+    let fields = directive.arguments.find((i: any) => i.name.value === "fields");
+    const fieldIndex = node.fields.findIndex((field: any) => field.name.value === fields.value.values[0].value);
+    let args: any[] = [];
+    if (fields.value.values.length !== 1) {
+        args = [createArgumentNode('sortKeyFields', {
+                ...fields.value,
+                values: fields.value.values.slice(1)
+            }
+        )];
+    }
+    node.fields[fieldIndex].directives.push(createDirectiveNode('primaryKey', args));
+}
+
+export function isSecondaryKey(directive: any) {
+    if (directive.name.value === 'key'
```

Comment:
Same comments here. You can use `some()` instead of `find()` and just return the result of the expression without the `if` statement.